lib/charms/opensearch/v0/helper_security.py

```diff
@@ -136,58 +143,147 @@ def split_ca_chain(pem_content: str) -> list[str]:
-            args = f"-storepass {store_pwd}"
             try:
-                run_cmd(cmd, args)
-                logger.info(
-                    f"Current CA {alias}-{index} was renamed to {OLD_CA_PREFIX}{alias}-{index}."
+                run_cmd(
+                    f"{KEYTOOL} -changealias "
+                    f"-alias {internal_alias} -destalias {old_internal_alias} "
+                    f"-keystore {store_path} -storetype PKCS12",
+                    f"-storepass {store_pwd}",
                 )
             except OpenSearchCmdError as e:
-                # This message means there was no "ca" alias or store before, if it happens ignore
-                if not (
-                    e.out is not None
-                    and (
-                        f"Alias <{alias}-{index}> does not exist" in e.out
-                        or "Keystore file does not exist" in e.out
-                    )
-                ):
-                    raise
-
-        with tempfile.NamedTemporaryFile(
-            mode="w+t", dir=os.path.dirname(store_path)
-        ) as ca_tmp_file:
-            ca_tmp_file.write(certs[index])
-            ca_tmp_file.flush()
+                msg = (e.out or "") + (e.err or "")
+                if ("does not exist" not in msg) and ("Keystore file does not exist" not in msg):
+                    return False
+
+        # import the cert
+        try:
+            with tempfile.NamedTemporaryFile(
+                dir=tmpdir,
+                mode="w",
+                encoding="utf-8",
+                errors="replace",
+                delete=False,
+            ) as tmp:
+                tmp.write(pem)
+                tmp.flush()
+                tmp_path = tmp.name
+
             try:
                 run_cmd(
-                    f"""{KEYTOOL} -importcert \
-                    -noprompt \
-                    -alias {alias}-{index} \
-                    -keystore {store_path} \
-                    -file {ca_tmp_file.name} \
-                    -storetype PKCS12
-                    """,
+                    f"{KEYTOOL} -importcert -noprompt "
+                    f"-alias {internal_alias} -keystore {store_path} -file {tmp_path} -storetype PKCS12",
                     f"-storepass {store_pwd}",
                 )
-                run_cmd(f"sudo chmod +r {store_path}")
-                logger.info("New CA was added to truststore.")
             except OpenSearchCmdError as e:
-                logger.error("Error storing the ca-cert: %s", e)
+                logger.error(
+                    "Failed to import cert for alias %s into %s: %s",
+                    internal_alias,
+                    store_path,
+                    (e.out or "") + (e.err or ""),
+                )
                 return False
+            finally:
+                # clean up temp file
+                with suppress(FileNotFoundError):
+                    os.remove(tmp_path)
```

Comment:
I think this defeats the purpose of having a context manager, why not either have `delete=True` or remove the context manager at all and reduce the level of nestedness by 1?